libopae/src/usrclk/user_clk_pgm_uclock.c

```diff
@@ -99,23 +99,35 @@ fpga_result __FIXME_MAKE_VISIBLE__ set_userclock(const char* sysfs_path,
 					uint64_t userclk_high,
 					uint64_t userclk_low)
 {
+	uint64_t refClock  = 0;
+
 	if (sysfs_path == NULL) {
-		FPGA_ERR("Invalid input parameters");
+		FPGA_ERR("Invalid Input parameters");
 		return FPGA_INVALID_PARAM;
 	}
 
+	// set low refclk if only one clock is avalible 
+	if ( userclk_high == 0 && userclk_low != 0 ) {
+		refClock = userclk_low;
+	}
+
 	// verify user clock freq range (100hz to 1200hz)
 	if ((userclk_high > MAX_FPGA_FREQ) ||
-		(userclk_high < MIN_FPGA_FREQ) ||
-		(userclk_low > (MAX_FPGA_FREQ / 2)) ||
-		(userclk_low < (MIN_FPGA_FREQ / 2))) {
+		(userclk_low > MAX_FPGA_FREQ)) {
+		FPGA_ERR("Invalid Input frequency");
+		return FPGA_INVALID_PARAM;
+	}
 
-		FPGA_ERR("Invalid input frequency");
+	if (userclk_low != 0  && userclk_low > userclk_high) {
```

Comment:
What about the case when userclk_low != 0 and userclk_high == 0 , like on line 110?